pkg/transportrequest/uploadcts_test.go

```diff
@@ -0,0 +1,156 @@
+package transportrequest
+
+import (
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestUploadCTS(t *testing.T) {
+
+	filesMock := mock.FilesMock{}
+	files = &filesMock
+	defer func() { files = piperutils.Files{} }()
```

Comment:
Would it not be better to reset the files after each t.Run?

Reply:
It would be technically possible to use a new file system mock in each test and I'm not strongly against that. But for most of the tests the file system mock does not matter since the case which is tested does not take the file system into account. For these cases the mock is just provided in order to decouple from the real file system on the machine running the tests. For the tests where the file mock matters there is already a new file system mock provided (respective the existing file system mock is used in the case that one fits for the test). I don't think the additional code which we have for resetting the file mock from each test is not justified. @o-liver  Do you think it would be?

Reply:
I understand. I agree that it is alright to leave as it is. These tests cannot run in parallel though, correct? But maybe that was never the case anyway with `files` being a global variable. I don't understand this file mocking entirely.